src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/CcdDashboardClaimMatcher.java

```diff
@@ -160,4 +165,88 @@ public boolean isHearingFormGenerated() {
         return !caseData.getHearingDocuments().isEmpty();
     }
 
+    @Override
+    public boolean sdoBeenDrawn() {
+        return caseData.getSystemGeneratedCaseDocuments().stream()
+            .anyMatch(systemGeneratedCaseDocument -> systemGeneratedCaseDocument
+                .getValue()
+                .getDocumentType().equals(DocumentType.SDO_ORDER));
+    }
+
+    @Override
+    public boolean beforeHearing() {
+        return caseData.getHearingDate().isAfter(LocalDateTime.now().toLocalDate());
+    }
+
+    @Override
+    public boolean isMoreDetailsRequired() {
+        return sdoBeenDrawn() && beforeHearing();
+    }
+
+    @Override
+    public boolean isMediationSuccessful() {
+        return !sdoBeenDrawn()
+            && !Objects.isNull(caseData.getMediation())
+            && !Objects.isNull(caseData.getMediation().getMediationSuccessful())
+            && !Objects.isNull(caseData.getMediation().getMediationSuccessful().getMediationAgreement());
+    }
+
+    @Override
+    public boolean isMediationUnsuccessful() {
+        return !sdoBeenDrawn()
+            && !Objects.isNull(caseData.getMediation())
+            && !Objects.isNull(caseData.getMediation().getUnsuccessfulMediationReason())
+            && !caseData.getMediation().getUnsuccessfulMediationReason().isEmpty();
+    }
+
+    @Override
+    public boolean isMediationPending() {
+        return !Objects.isNull(caseData.getCcdState())
+            && caseData.getCcdState().equals(CaseState.IN_MEDIATION)
+            && !Objects.isNull(caseData.getMediation())
+            && !Objects.isNull(caseData.getMediation().getMediationSuccessful())
+            && Objects.isNull(caseData.getMediation().getMediationSuccessful().getMediationAgreement());
+    }
+
+    @Override
+    public boolean isCourtReviewing() {
+        return !sdoBeenDrawn()
+            && !Objects.isNull(caseData.getApplicant1ProceedsWithClaimSpec())
+            && caseData.getApplicant1ProceedsWithClaimSpec().equals(YesOrNo.YES)
+            && caseData.isRespondentResponseFullDefence()
+            && caseData.hasApplicantRejectedRepaymentPlan();
+    }
+
+    @Override
+    public boolean isClaimEnded() {
+        return !Objects.isNull(caseData.getApplicant1ProceedsWithClaimSpec())
```

Comment:
same here

Reply:
also i think we have a method in claim that checks whether the applicant proceeds with claim. please have a look and use that instead

Reply:
sorry that is a little bit confusing, because I am using here `getApplicant1PRoccedsWithClaimSpec` method and bellow I am checking if yes, I did not see method which doing that for me